File: leetcode/Arrays/leetcode-0228/228.java
class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> res = new ArrayList<>();
        if (nums == null || nums.length == 0) {
            return res;
        }

        int start = nums[0], end = nums[0];
        for (int i = 1; i < nums.length; i++) {
            if (nums[i] - end == 1) {
                end = nums[i];
                continue;
            } else {
                String tmp = getStr(start, end);
                res.add(tmp);
                start = nums[i];
                end = nums[i];
            }
        }
        String tmp = getStr(start, end);
        res.add(tmp);
        return res;
    }

    private String getStr(int start, int end) {
        if (start == end) {
            return start + "";
        }
        return start + "->" + end;
    }
}

File: leetcode/Arrays/leetcode-0228/228.go
package arrays

func summaryRanges(nums []int) []string {
	res := []string{}

	if nums == nil || len(nums) == 0 {
		return res
	}

	start, end := nums[0], nums[0]

	for i := 1; i < len(nums); i++ {
		if nums[i]-end == 1 {
			end = nums[i]
			continue
		} else {
			tmp := getStr(start, end)
			res = append(res, tmp)
			start = nums[i]
			end = nums[i]
		}
	}
	tmp := getStr(start, end)
	res = append(res, tmp)
	return res
}

func getStr(start, end int) string {
	if start == end {
		return strconv.Itoa(start)
	}
	return strconv.Itoa(start) + "->" + strconv.Itoa(end)
}
